packages/jest-types/src/Config.ts

```diff
@@ -12,7 +12,104 @@ import type {SnapshotFormat} from '@jest/schemas';
 
 type CoverageProvider = 'babel' | 'v8';
 
-type Timers = 'real' | 'fake' | 'modern' | 'legacy';
+export type FakeableTimerAPIs =
+  | 'setImmediate'
+  | 'clearImmediate'
+  | 'setInterval'
+  | 'clearInterval'
+  | 'setTimeout'
+  | 'clearTimeout'
+  | 'requestAnimationFrame'
+  | 'cancelAnimationFrame'
+  | 'requestIdleCallback'
+  | 'cancelIdleCallback'
+  | 'Date'
+  | 'hrtime'
+  | 'nextTick'
+  | 'performance';
+// | 'queueMicrotask'; // ?
+
+export type LegacyTimersConfig = {
+  /**
+   * The strategy to be use for the fake timers.
+   *
+   * @defaultValue
+   * The default is `'modern'`.
+   */
+  strategy?: 'legacy';
+  /**
+   * The maximum number of timers that will be run when calling `jest.advanceTimersByTime()`,
+   * `jest.runAllImmediates()`, `jest.runAllTicks()` or `jest.runAllTimers()`.
+   *
+   * @defaultValue
+   * The default is `100_000` timers.
+   */
+  loopLimit?: number;
+};
+
+export type ModernTimersConfig = {
+  /**
+   * The strategy to be use for the fake timers.
+   *
+   * @defaultValue
+   * The default is `'modern'`.
+   */
+  strategy?: 'modern';
+  /**
+   * Sets the default unix epoch. May be a number (in milliseconds) or a Date object.
+   *
+   * @defaultValue
+   * The default is `Date.now()`.
+   */
+  now?: number | Date;
+  /**
+   * Allows to cherry pick the timer methods and objects (e.g. `setTimeout()`,
+   * `setImmediate()`, `Date`, `nextTick`, `performance`) that should be faked.
+   *
+   * @defaultValue
+   * All timer APIs are faked by default.
+   *
+   * @example
+   * {toFake: ['nextTick', 'setTimeout']}
+   */
+  toFake?: Array<FakeableTimerAPIs>;
```

Comment:
Perhaps a counterpart option `notToFake` (with better name?) could be added as well. See #11661

Reply:
Hm.. Since the default in Jest is to fake all the APIs, it might be even better to have something like `doNotFake?: Array<FakeableTimerAPIs>` instead of `toFake`.